src/common/tvgEnv.h

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) 2025 the ThorVG project. All rights reserved.
+
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#ifndef _TVG_ENV_H_
+#define _TVG_ENV_H_
+
+#ifdef _WIN32
+    #ifndef WIN32_LEAN_AND_MEAN
+        #define WIN32_LEAN_AND_MEAN
+        #include <windows.h>
+    #endif
+    #ifndef PATH_MAX
+        #define PATH_MAX MAX_PATH
+    #endif
+#else
+    #include <limits.h>
+    #include <sys/stat.h>
+    #include <unistd.h>
+#endif
+
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
+
+namespace tvg
+{
+    inline bool getCacheDir(char* cacheDir, size_t maxLen)
+    {
+#if !defined(THORVG_FILE_IO_SUPPORT) || defined(__EMSCRIPTEN__)
+        return false;
+#endif
+
+#ifdef _WIN32
+        // Windows: %LOCALAPPDATA%\thorvg\shaders
+        auto appData = getenv("LOCALAPPDATA");
+        if (!appData) {
+            appData = getenv("APPDATA");
+            if (!appData) return false;
+        }
+        snprintf(cacheDir, maxLen, "%s\\thorvg\\shaders", appData);
+
+        auto attribs = GetFileAttributesA(cacheDir);
+        if (attribs == INVALID_FILE_ATTRIBUTES) {
+            char thorvgDir[PATH_MAX];
+            snprintf(thorvgDir, PATH_MAX, "%s\\thorvg", appData);
+
+            if (!CreateDirectoryA(thorvgDir, NULL) && GetLastError() != ERROR_ALREADY_EXISTS) return false;
+
+            if (!CreateDirectoryA(cacheDir, NULL) && GetLastError() != ERROR_ALREADY_EXISTS) return false;
+        }
+#elif defined(__linux__)
+        // Linux: $XDG_CACHE_HOME/thorvg/shaders or ~/.cache/thorvg/shaders
+        auto xdgCache = getenv("XDG_CACHE_HOME");
+        if (xdgCache) {
+            snprintf(cacheDir, maxLen, "%s/thorvg/shaders", xdgCache);
+        } else {
+            auto home = getenv("HOME");
+            if (!home) return false;
+            snprintf(cacheDir, maxLen, "%s/.cache/thorvg/shaders", home);
```

Comment:
This function needs to have `%s/.cache/thorvg/`

2 options, we can have:
1. you can just put the shader files in thorvg without `shaders`
2. you can append the `shaders` subfolder from gl engine

I bet on no.1